### PR TITLE
set a minimal size in the compression output buffer

### PR DESCRIPTION
**Pull request by Geal**
it should be at least large enough to hold the 10 byte gzip header. Otherwise the compressor goes into an infinite loop

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Review by BrynCooke — CHANGES_REQUESTED** [review]
Can we have a test for compression?

**Comment by Geal** [comment]
there's a test now

**Review by BrynCooke — APPROVED** [review]

File: apollo-router/src/axum_factory/compression/mod.rs [hunk 1]
```diff
@@ -79,7 +79,9 @@ where {
                         }
                     }
                     Ok(data) => {
-                        let mut buf = BytesMut::zeroed(data.len());
+                        // the buffer needs at least 10 bytes for a gzip header if we use gzip, then more
+                        // room to store the data itself
+                        let mut buf = BytesMut::zeroed(10 + data.len());
```
Comment (garypen) [inline apollo-router/src/axum_factory/compression/mod.rs:84]: 10 should be a const GZIP_HEADER_LEN.

Comment (Geal) [reply apollo-router/src/axum_factory/compression/mod.rs]: done in [374a868](https://github.com/apollographql/router/pull/3961/commits/374a8688cb6b34b927af839388e3dafd8335e72d)

**Pull request merged**